demo/admin/src/pages/blocks/FullWidthImageBlock.tsx

```diff
@@ -9,19 +9,37 @@ const FullWidthImageContentBlock = createOptionalBlock(RichTextBlock, {
     title: <FormattedMessage {...messages.content} />,
 });
 
-export const FullWidthImageBlock = createCompositeBlock({
-    name: "FullWidthImage",
-    displayName: <FormattedMessage id="blocks.fullWidthImage" defaultMessage="Full Width Image" />,
-    category: BlockCategory.Media,
-    blocks: {
-        image: {
-            block: DamImageBlock,
-            title: <FormattedMessage {...messages.image} />,
-            paper: true,
-        },
-        content: {
-            block: FullWidthImageContentBlock,
-            title: <FormattedMessage {...messages.content} />,
+export const FullWidthImageBlock = createCompositeBlock(
+    {
+        name: "FullWidthImage",
+        displayName: <FormattedMessage id="blocks.fullWidthImage" defaultMessage="Full Width Image" />,
+        category: BlockCategory.Media,
+        blocks: {
+            image: {
+                block: DamImageBlock,
+                title: <FormattedMessage {...messages.image} />,
+                paper: true,
+            },
+            content: {
+                block: FullWidthImageContentBlock,
+                title: <FormattedMessage {...messages.content} />,
+            },
         },
     },
-});
+    (block) => ({
+        ...block,
+        extractTextContents: (block) => {
+            const content = block.content?.block ? RichTextBlock.extractTextContents?.(block.content?.block ?? {}) ?? [] : [];
+            return content;
+        },
+        replaceTextContents: (state, contents) => ({
+            ...state,
+            content: {
+                ...state.content,
+                block: state.content.block
+                    ? RichTextBlock.replaceTextContents?.(state.content.block, contents) ?? state.content.block
+                    : state.content.block,
+            },
```

Comment:
You should be able to call `FullWidthImageContentBlock.replaceTextContents` here